### ISSUE TITLE
Under PopOS, docker run works, while gitlab-ci-local doesn't 

### ISSUE THREAD

**Issue by jg-cantaa** [firecow/gitlab-ci-local#361]
## TL;DR: This Issue happens under PopOS. The cause could not be pinpointed despite intensive debugging, so this Issue will stay unfixed. Currently the only workaround is to run kaniko with the `--force` flag. Using `gitlab-ci-local --privileged` will not work.

![2022-01-14_10-30](https://user-images.githubusercontent.com/95756912/149492986-a013adda-581f-4e0a-9681-d23f3e286952.png)

**Minimal Project Structure Archive:**
[Issue.zip](https://github.com/firecow/gitlab-ci-local/files/7718797/Issue.zip)

**Expected behavior**
Like run.sh: the container builds successfully

**Host information**
Linux: Pop!_OS 20.04 LTS
gitlab-ci-local 4.27.1

**Note**
This issue requires the same setup like #341, however the error is a new one.

**Comment by firecow** [comment]
@jg-cantaa https://github.com/firecow/gitlab-ci-local/releases/tag/4.28.1 Can you try out this version please, and reopen if the bug still persists.
Works on my machine, with 4.28.1

**Comment by jg-cantaa** [comment]
Sadly the Issue persists, even with 4.28.1. 
![2022-01-18_11-38](https://user-images.githubusercontent.com/95756912/149921585-4ef2f991-0e8d-45ac-b510-982b2ce5d250.png)

I have tried removing the dockerimages to redownload them. However I don't know what else I could try in order to get it working. Please let me know if you want me to test other workarounds/fixes.

**Comment by firecow** [comment]
@jg-cantaa Hmm.. Why the hell can't I reproduce this...
What docker version are you running ?

**Comment by jg-cantaa** [comment]
Docker version 20.10.7, build 20.10.7-0ubuntu5.1

**Comment by jg-cantaa** [comment]
Yes it does, but that kind of defeats the purpose of using gitlab-ci-local. It should just work. It should not require --force, since a normal docker container also does not require --force. 
Now, It could also be the case, that it's not worth it to fix this, simply because Kaniko requires some obscure thing to indicate it running in a container, which might not be that easy to reproduce.

**Comment by firecow** [comment]
> Yes it does, but that kind of defeats the purpose of using gitlab-ci-local. It should just work. It should not require --force, since a normal docker container also does not require --force. Now, It could also be the case, that it's not worth it to fix this, simply because Kaniko requires some obscure thing to indicate it running in a container, which might not be that easy to reproduce.

This shall be fixed, lots of people uses Kaniko.

Can you try my example real quick.

**Comment by jg-cantaa** [comment]
Still... 
![2022-01-18_14-29](https://user-images.githubusercontent.com/95756912/149946396-387a10e0-6e86-415a-ad98-925cf6ec0a51.png)

**Comment by firecow** [comment]
![image](https://user-images.githubusercontent.com/973602/149946938-99ec628c-07c5-466a-8b43-ffc13b0a8f06.png)

This is mine @jg-cantaa 

@bcouetil Can you try running the `test:` job from above as well

**Comment by jg-cantaa** [comment]
Are there any other programs with specific versions or packages that could influence this kind of behavior?

**Comment by firecow** [comment]
Do you have `.gitlab-ci-local-env` or `.gitlab-ci-local.yml` located in CWD @jg-cantaa ?

**Comment by firecow** [comment]
@jg-cantaa Ok, make a new dir, and only put a `.gitlab-ci.yml` with the `test:` inside and call `gitlab-ci-local test`
Just so we are 2000% sure nothing file/folder specific is fucking us.

**Comment by firecow** [comment]
Why are we talking about ts-node -v now ? @jg-cantaa

**Comment by firecow** [comment]
@jg-cantaa Yes, it could become useful later on.

Right now, i just need to see output similar to this. When moving to a new non-git folder

![image](https://user-images.githubusercontent.com/973602/149950281-528a4c65-09bf-4b5c-a29d-227d8ab1a166.png)

**Comment by jg-cantaa** [comment]
![2022-01-18_14-57](https://user-images.githubusercontent.com/95756912/149951028-a4fd8fa2-2435-49e2-b91a-ecd21daa3ae7.png)

**Comment by firecow** [comment]
```
mjn@mjn-laptop:~$ cat /etc/docker/daemon.json 
{
  "metrics-addr" : "127.0.0.1:9323",
  "experimental" : true,
  "registry-mirrors": ["https://mirror.gcr.io"]
}
```

Here is mine, what's yours :laughing:

**Comment by bcouetil** [comment]
It might be a PopOS issue...

**Comment by firecow** [comment]
Hmm.... I'm pretty sure I have a dev that uses PopOS, I'll try his machine in the morrow 😁

**Comment by firecow** [comment]
![image](https://user-images.githubusercontent.com/973602/151329799-99068258-5ce6-4200-8c9b-f4c97ce2d953.png)

It really looks like something todo with PopOS now :smile: 

My PopOS co-worker is getting it.

**Comment by firecow** [comment]
@jg-cantaa Can you print your bash version please
```
bash --version
```

**Comment by jg-cantaa** [comment]
```
~$ bash --version
GNU bash, Version 5.1.8(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2020 Free Software Foundation, Inc.
Lizenz GPLv3+: GNU GPL Version 3 oder jünger <http://gnu.org/licenses/gpl.html>

Dies ist freie Software.  Sie darf verändert und verteilt werden.
Es wird keine Garantie gewährt, soweit das Gesetz es zulässt.
```

**Comment by firecow** [comment]
@jg-cantaa Would you by any chance have `selinux` installed ?

**Comment by jg-cantaa** [comment]
No.
```
~$ selinux
selinux: Befehl nicht gefunden.

```

**Comment by firecow** [comment]
Try `apt list *selinux*`

Also does `gitlab-ci-local --priviledged` get rid of the warning?

**Comment by jg-cantaa** [comment]
So, apparently I do...
```
~$ apt list *selinux*
Auflistung… Fertig
android-libselinux-dev/impish 10.0.0+r36-1 amd64
android-libselinux/impish 10.0.0+r36-1 amd64
golang-github-opencontainers-selinux-dev/impish,impish 1.8.0-1 all
libselinux1-dev/impish,now 3.1-3build2 amd64  [Installiert,automatisch]
libselinux1-dev/impish 3.1-3build2 i386
libselinux1/impish,now 3.1-3build2 amd64  [installiert]
libselinux1/impish 3.1-3build2 i386
puppet-module-puppetlabs-selinux-core/impish,impish 1.0.4-2 all
python3-selinux/impish 3.1-3build2 amd64
python3-selinux/impish 3.1-3build2 i386
ruby-selinux/impish 3.1-3build2 amd64
ruby-selinux/impish 3.1-3build2 i386
selinux-basics/impish,impish 0.5.8 all
selinux-policy-default/impish,impish 2:2.20210203-7 all
selinux-policy-dev/impish,impish 2:2.20210203-7 all
selinux-policy-doc/impish,impish 2:2.20210203-7 all
selinux-policy-mls/impish,impish 2:2.20210203-7 all
selinux-policy-src/impish,impish 2:2.20210203-7 all
selinux-utils/impish 3.1-3build2 amd64
selinux-utils/impish 3.1-3build2 i386
```
Using `--privleged` also does not remove the warning:
![2022-02-04_09-52](https://user-images.githubusercontent.com/95756912/152499773-c2061113-24d1-4d78-bca5-09ee346d562d.png)

**Comment by jg-cantaa** [comment]
Thats totally ok! Thanks a lot for the effort you put in. It's sad that the cause could not be found.

**Comment by firecow** [comment]
@jg-cantaa Can you try to reproduce, using the 4.31.0, my coworker doesn't seem to be experiencing the problem anymore, and I have no idea why :smile:

**Comment by jg-cantaa** [comment]
I tried again and it still happens. 
(I took the issue.zip from above, fixed the script that was apparently faulty and added a git repo so gitlab-ci-local does not complain: [kaniko.zip](https://github.com/firecow/gitlab-ci-local/files/8797597/kaniko.zip))

**Comment by firecow** [comment]
@jg-cantaa Dang it.. But thanks for the try...

**Comment by firecow** [comment]
@jg-cantaa Can you try the latest version of `gitlab-ci-local`

I heard rumors around the office that we have several developers using gitlab-ci-local under PopOS.

**Comment by jg-cantaa** [comment]
Soo, apparently it just works now. 
I tried with and without the --force flag in the gitlab-ci.yml and it worked both times.

Thanks for the effort :)
